src/pspm_gaze_pp.m

```diff
@@ -0,0 +1,202 @@
+function [sts, out_channel] = pspm_gaze_pp(fn, options)
+
+% ●	Description
+% 	pspm_gaze_pp preprocesses gaze signals, gaze x and gaze y channels at
+% 	the same time.
+% ●	Format
+% 	[sts, out_channel] = pspm_gaze_pp(fn) or
+% 	[sts, out_channel] = pspm_gaze_pp(fn, options)
+% ●	Arguments
+% 	              fn: [string] Path to the PsPM file which contains the gaze data.
+% 	         options: [struct]
+%           .channel: [numeric/string, optional] Channel ID to be preprocessed.
+%   .channel_combine: [numeric/string, optional] Channel ID to be combined.
+%      .valid_sample: [bool] 1 or 0. 1 if use valid samples produced by
+%                     pspm_pupil_pp, 0 if not to use. default as 0.
+% ●	Authors
+% 	(C) 2021 Teddy Chao (UCL)
+
+%% 1 Initialise
+global settings;
+if isempty(settings)
+  pspm_init;
+end
+sts = -1;
+%% 2 Create default arguments
+% 2.1 static variables
+list_channels = {'gaze_x_l', 'gaze_x_r', 'gaze_y_l', 'gaze_y_r'};
+% 2.2 set default values
+if nargin == 1;                        options = struct();              end
+if ~isfield(options,'channel');        options.channel = 'gaze_x_l';    end
+if ~isfield(options,'channel_action'); options.channel_action = 'add';  end
+if ~isfield(options,'channel_combine');options.channel_combine = 'none';end
+if ~isfield(options,'valid_sample');   options.valid_sample = 0;        end
+if ~isfield(options, 'plot_data');     options.plot_data = false;       end
+if ~isfield(options, 'segments');      options.segments = {};           end
+action_combine = ~strcmp(options.channel_combine, 'none');
+% 2.3 set default options from pupil_pp
+[lsts, default_settings] = pspm_pupil_pp_options();
+if lsts ~= 1
+  return
+end
+if isfield(options, 'custom_settings')
+  default_settings = pspm_assign_fields_recursively( ...
+    default_settings, options.custom_settings);
+end
+options.custom_settings = default_settings;
+%% 3 Input checks
+% 3.1 check the file exist
+[sts, ~, ~, ~] = pspm_load_data(fn);
+if sts ~= 1; warning('ID:invalid_input', 'cannot load data from the file'); return; end
```

Comment:
> Are you sure an error wouldn't be more appropriate here ?

We can discuss with Dom if we want to change "warning(), ... return" to "error()" since currently all PsPM functions are using the former option.